ironfish/src/networkBridge.ts

```diff
@@ -102,43 +103,46 @@ export class NetworkBridge {
       })
     })
 
-    this.node.captain.onRequestBlocks.on((hash: BlockHash, nextBlockDirection: boolean) => {
-      Assert.isNotNull(this.node)
-      Assert.isNotNull(this.node.captain)
-      Assert.isNotNull(this.peerNetwork)
-      Assert.isNotNull(this.node)
-
-      const serializedHash = this.node.captain.chain.blockHashSerde.serialize(hash)
-
-      const request: BlockRequest = {
-        type: MessageType.Blocks,
-        payload: {
-          hash: serializedHash,
-          nextBlockDirection: nextBlockDirection,
-        },
-      }
-
-      this.peerNetwork
-        .request(request)
-        .then((c) => {
-          if (
-            !c ||
-            !isBlocksResponse<SerializedWasmNoteEncrypted, SerializedTransaction>(c.message)
-          ) {
-            throw new Error('Invalid format')
-          }
-          this.onBlockResponses(
-            {
-              ...c,
-              message: c.message,
-            },
-            request,
-          )
-        })
-        .catch((err) => {
-          this.node?.captain?.blockSyncer.handleBlockRequestError(request, err)
-        })
-    })
+    this.node.captain.onRequestBlocks.on(
+      (hash: BlockHash, nextBlockDirection: boolean, peer?: Identity) => {
+        Assert.isNotNull(this.node)
+        Assert.isNotNull(this.node.captain)
+        Assert.isNotNull(this.peerNetwork)
+        Assert.isNotNull(this.node)
+
+        const serializedHash = this.node.captain.chain.blockHashSerde.serialize(hash)
+
+        const request: BlockRequest = {
+          type: MessageType.Blocks,
+          payload: {
+            hash: serializedHash,
+            nextBlockDirection: nextBlockDirection,
+          },
+          peer: peer,
```

Comment:
if we pass this as an argument to request, we won't need to modify the core messagetype to contain this

Reply:
related, I think this will end up sending the peer identity along with the message in `request`